design/one-pager-simple-class-selection.md

```diff
@@ -0,0 +1,411 @@
+resource classes] for details.
+
+## Goals
+
+The goal of this proposal is to __make it easier for potential Crossplane users
+to learn and take advantage of the portability and separation of concerns__ that
+are enabled by dynamic provisioning. A _secondary_ goal is to maintain the
+flexibility and power that Crossplane offers complex organisations with many
+application and infrastructure namespaces.
+
+## Proposal
+
+This document proposes Crossplane retire the concept of portable resource
+classes. Resource claims would instead be matched directly to non-portable
+resource classes (referred to henceforth as simply 'resource classes') using
+[label selectors]. Providers, managed resources, and resource classes would be
+cluster scoped rather than namespaced.
+
+```yaml
+apiVersion: database.crossplane.io/v1alpha1
+kind: PostgreSQLInstance
+metadata:
+  namespace: acme-team
+  name: gitlab-database
+spec:
+  classSelector:
+    matchLabels:
+      stack: gitlab
+      grade: experimental
+      region: us-east-1
+```
+
+Consider the hypothetical resource claim above, created by an equally
+hypothetical GitLab [stack]. It declares a need for an experimental grade
+PostgreSQL instance in region us-east-1 that is suitable for use with the GitLab
+stack. Any compatible resource class (whether it be an `RDSInstanceClass` or a
+`CloudSQLInstanceClass`) matching the resource claim's labels would be eligible
+to satisfy it. If more than one such class existed one would be selected at
+random.
+
+This pattern is likely familiar to Kubernetes and to Crossplane users - it's
+used to select which nodes a pod may run on, which pods are endpoints of a
+service, and which Kubernetes cluster a Crossplane workload may run on.
+
+### Infrastructure Scope
+
+Crossplane 0.3 introduced the concept of an 'infrastructure namespace' - a
+logical grouping of infrastructure resources such as providers, resource
+classes, and managed resources. These namespaces might group infrastructure by
+its geographical region, or its environment (e.g. "production"), distinct from
+application namespaces. This potentially many to many application to
+infrastructure namespace pattern - currently established by portable resource
+classes - makes it difficult to mentally model the relationship between
+applications and the infrastructure that supports them.
+
+It is an uncommon pattern in Kubernetes to namespace infrastructure resources.
+Existing infrastructure kinds such as nodes, persistent volumes, and storage
+classes are cluster scoped concepts. This document proposes Crossplane Services
+resources (with the exception of resource claims) become cluster scoped. This
+provides a single, easy to find scope in which resource claim authors can
+discover the resource classes available to them, and the managed resources
+produced by their claim. The logical groupings provided by infrastructure
+namespaces that we frequently think of as “production”, “development”, or
```

Comment:
one consideration here is that infra namespaces support simpler RBAC for a prod-infra vs. dev-infra namespaces, reducing the likelihood of a shared cluster / control plane environments where one group can define the dev/test resource classes to experiment with new settings, and a more limited group has access to prod-infra.

Reply:
This is true - updated the doc to capture this.